[CI] CanMatchPreFilterSearchPhaseTests testCanMatchFilteringOnCoordinatorThatCanBeSkipped failing

I can't reproduce it locally but it could potentially be a deterministic failure. Same error also already occurred two weeks ago: https://gradle-enterprise.elastic.co/scans/tests?search.relativeStartTime=P28D&search.timeZoneId=Europe/Zurich&tests.container=org.elasticsearch.action.search.CanMatchPreFilterSearchPhaseTests&tests.sortField=FAILED&tests.test=testCanMatchFilteringOnCoordinatorThatCanBeSkipped&tests.unstableOnly=true

**Build scan:**
https://gradle-enterprise.elastic.co/s/3lhprsffledqm/tests/:server:test/org.elasticsearch.action.search.CanMatchPreFilterSearchPhaseTests/testCanMatchFilteringOnCoordinatorThatCanBeSkipped

**Reproduction line:**
`./gradlew ':server:test' --tests "org.elasticsearch.action.search.CanMatchPreFilterSearchPhaseTests.testCanMatchFilteringOnCoordinatorThatCanBeSkipped" -Dtests.seed=121232E157FE3BFF -Dtests.locale=ca -Dtests.timezone=Etc/GMT-14 -Druntime.java=8`

**Applicable branches:**
7.14

**Reproduces locally?:**
No

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.action.search.CanMatchPreFilterSearchPhaseTests&tests.test=testCanMatchFilteringOnCoordinatorThatCanBeSkipped

**Failure excerpt:**
```
java.lang.AssertionError: 
Expected: <true>
     but: was <false>

  at __randomizedtesting.SeedInfo.seed([121232E157FE3BFF:F1A710FFEF061B4E]:0)
  at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
  at org.junit.Assert.assertThat(Assert.java:956)
  at org.junit.Assert.assertThat(Assert.java:923)
  at org.elasticsearch.action.search.CanMatchPreFilterSearchPhaseTests.lambda$testCanMatchFilteringOnCoordinatorThatCanBeSkipped$21(CanMatchPreFilterSearchPhaseTests.java:498)
  at org.elasticsearch.action.search.CanMatchPreFilterSearchPhaseTests.assignShardsAndExecuteCanMatchPhase(CanMatchPreFilterSearchPhaseTests.java:754)
  at org.elasticsearch.action.search.CanMatchPreFilterSearchPhaseTests.testCanMatchFilteringOnCoordinatorThatCanBeSkipped(CanMatchPreFilterSearchPhaseTests.java:476)
  at sun.reflect.NativeMethodAccessorImpl.invoke0(NativeMethodAccessorImpl.java:-2)
  at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
  at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  at java.lang.reflect.Method.invoke(Method.java:498)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1750)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:938)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:974)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:988)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:49)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
  at org.apache.lucene.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:48)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:817)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:468)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:947)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:832)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:883)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:894)
  at org.apache.lucene.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:45)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:41)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:47)
  at org.apache.lucene.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:64)
  at org.apache.lucene.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:54)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:368)
  at java.lang.Thread.run(Thread.java:748)

```

--- a/server/src/test/java/org/elasticsearch/action/search/CanMatchPreFilterSearchPhaseTests.java
+++ b/server/src/test/java/org/elasticsearch/action/search/CanMatchPreFilterSearchPhaseTests.java
@@ -451,7 +451,7 @@ public class CanMatchPreFilterSearchPhaseTests extends ESTestCase {
         List<Index> regularIndices =
             randomList(0, 2, () -> new Index(randomAlphaOfLength(10), UUIDs.base64UUID()));
 
-        long indexMinTimestamp = randomLongBetween(0, 5000);
+        long indexMinTimestamp = randomLongBetween(1000, 5000);
         long indexMaxTimestamp = randomLongBetween(indexMinTimestamp, 5000 * 2);
         StaticCoordinatorRewriteContextProviderBuilder contextProviderBuilder = new StaticCoordinatorRewriteContextProviderBuilder();
         String timestampFieldName = dataStream.getTimeStampField().getName();
@@ -463,7 +463,8 @@ public class CanMatchPreFilterSearchPhaseTests extends ESTestCase {
         // We query a range outside of the timestamp range covered by both datastream indices
         rangeQueryBuilder
             .from(indexMaxTimestamp + 1)
-            .to(indexMaxTimestamp + 2);
+            .to(indexMaxTimestamp + 2)
+            .format("epoch_millis");
 
         BoolQueryBuilder queryBuilder = new BoolQueryBuilder()
             .filter(rangeQueryBuilder);
@@ -519,7 +520,7 @@ public class CanMatchPreFilterSearchPhaseTests extends ESTestCase {
         List<Index> regularIndices =
             randomList(0, 2, () -> new Index(randomAlphaOfLength(10), UUIDs.base64UUID()));
 
-        long indexMinTimestamp = randomLongBetween(0, 5000);
+        long indexMinTimestamp = randomLongBetween(1000, 5000);
         long indexMaxTimestamp = randomLongBetween(indexMinTimestamp, 5000 * 2);
         StaticCoordinatorRewriteContextProviderBuilder contextProviderBuilder = new StaticCoordinatorRewriteContextProviderBuilder();
         String timestampFieldName = dataStream.getTimeStampField().getName();
@@ -577,7 +578,8 @@ public class CanMatchPreFilterSearchPhaseTests extends ESTestCase {
             // We query a range within the timestamp range covered by both datastream indices
             rangeQueryBuilder
                 .from(indexMinTimestamp)
-                .to(indexMaxTimestamp);
+                .to(indexMaxTimestamp)
+                .format("epoch_millis");
 
             queryBuilder.filter(rangeQueryBuilder);
 
@@ -590,7 +592,8 @@ public class CanMatchPreFilterSearchPhaseTests extends ESTestCase {
             // We query a range outside of the timestamp range covered by both datastream indices
             RangeQueryBuilder rangeQueryBuilder = new RangeQueryBuilder(timestampFieldName)
                 .from(indexMaxTimestamp + 1)
-                .to(indexMaxTimestamp + 2);
+                .to(indexMaxTimestamp + 2)
+                .format("epoch_millis");
 
             TermQueryBuilder termQueryBuilder = new TermQueryBuilder("fake", "value");
 

